apps/console/src/features/users/components/add-user-updated.tsx

```diff
@@ -0,0 +1,947 @@
+    const [ userstore, setUserstore ] = useState<string>(PRIMARY_USERSTORE);
+    const [ isValidEmail, setIsValidEmail ] = useState<boolean>(false);
+    const [ isEmailFilled, setIsEmailFilled ] = useState<boolean>(false);
+
+    const formBottomRef: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>();
+    const emailRef: MutableRefObject<HTMLDivElement> = useRef<HTMLDivElement>();
+
+    const { t } = useTranslation();
+
+    const [ , setRegExLoading ] = useState<boolean>(false);
+
+    // Username input validation error messages.
+    const USER_ALREADY_EXIST_ERROR_MESSAGE: string = t("console:manage.features.users.consumerUsers.fields." +
+        "username.validations.invalid");
+    const USERNAME_REGEX_VIOLATION_ERROR_MESSAGE: string = t("console:manage.features.users.consumerUsers.fields." +
+        "username.validations.regExViolation");
+    const USERNAME_HAS_INVALID_CHARS_ERROR_MESSAGE: string = t("console:manage.features.users.consumerUsers.fields." +
+        "username.validations.invalidCharacters");
+    const USERNAME_HAS_INVALID_SYMBOLS_ERROR_MESSAGE: string = t("extensions:manage.features.user.addUser.validation." +
+        "usernameSymbols");
+    const USERNAME_HAS_INVALID_LENGTH_ERROR_MESSAGE: string =
+        t("extensions:manage.features.user.addUser.validation.usernameLength", {
+            maxLength: usernameConfig?.maxLength,
+            minLength: usernameConfig?.minLength
+        });
+
+    const eventPublisher: EventPublisher = EventPublisher.getInstance();
+
+    /**
+     * Set the password setup option to 'create-password'.
+     */
+    useEffect(() => {
+        if (!passwordOption) {
+            if (!requestedPasswordOption) {
+                setPasswordOption("create-password");
+
+                return;
+            }
+
+            setPasswordOption(requestedPasswordOption);
+        }
+    }, [ requestedPasswordOption ]);
+
+    useEffect(() => {
+        if (passwordOption === "create-password") {
+            setUserSummaryEnabled(true);
+            setAskPasswordFromUser(true);
+            setOfflineUser(false);
+        } else {
+            if (askPasswordOption === "offline") {
+                setUserSummaryEnabled(true);
+                setAskPasswordFromUser(false);
+                setOfflineUser(true);
+
+                return;
+            }
+            setUserSummaryEnabled(false);
+            setAskPasswordFromUser(false);
+            setOfflineUser(false);
+        }
+    }, [ passwordOption, askPasswordOption ]);
+
+    useEffect(() => {
+        if (validationConfig) {
+            setPasswordConfig(getConfiguration(validationConfig));
+            setUsernameConfig(getUsernameConfiguration(validationConfig));
+        }
+    }, [ validationConfig ]);
+
+    useEffect(() => {
+        if (!isAlphanumericUsernameEnabled()) {
+            setIsValidEmail(true);
+        } else {
+            setIsValidEmail(false);
+            // setAskPasswordOption("offline");
```

Comment:
Let's remove this if unused

Reply:
Fixed with fd5839362baf8e4e50397013a20cfa8b0b15fa4e